File: master_wallet/ios/TigerMasterWallet/Sources/Services/SuperAdminService.swift
// MasterWallet Super Admin Service (iOS)
//
// FAIL-CLOSED: SuperAdmin is an Admin-app feature, NOT a MasterWallet feature.
// The MasterWallet Go backend (:8450) does not expose any `/api/super-admin/*`
// routes (isolation guarantee). Therefore every method in this service refuses
// to make a network call and throws instead. The file is kept for parity with
// the rest of the client surface; no method is permitted to reach the backend.

import Foundation

/// Error thrown by every `SuperAdminService` method. Carries a fixed message
/// so callers can distinguish "feature not available here" from real backend
/// errors raised by `MasterAPIService`.
struct SuperAdminUnavailableError: Error, LocalizedError {
    var errorDescription: String? {
        "SuperAdmin is an Admin-app feature, not available in MasterWallet"
    }
}

class SuperAdminService {

    private let unavailableMessage = "SuperAdmin is an Admin-app feature, not available in MasterWallet"

    // MARK: - Initialize

    func initialize() throws {
        throw SuperAdminUnavailableError()
    }

    // MARK: - Authentication

    func authenticate(email: String, password: String) async throws {
        throw SuperAdminUnavailableError()
    }

    func logout() throws {
        throw SuperAdminUnavailableError()
    }

    // MARK: - Feature Flags

    func setFeatureFlag(name: String, enabled: Bool) throws {
        throw SuperAdminUnavailableError()
    }

    func getFeatureFlag(name: String) throws -> [String: Any] {
        throw SuperAdminUnavailableError()
    }

    func listFeatureFlags() throws -> [String: [String: Any]] {
        throw SuperAdminUnavailableError()
    }

    func isFeatureEnabled(name: String) throws -> Bool {
        throw SuperAdminUnavailableError()
    }

    // MARK: - Admin Management

    func listAdmins(roleFilter: String? = nil) async throws -> [[String: Any]] {
        throw SuperAdminUnavailableError()
    }

    // MARK: - Audit Logs

    func getAuditLogs(adminId: String? = nil, action: String? = nil, limit: Int = 100) async throws -> [[String: Any]] {
        throw SuperAdminUnavailableError()
    }

    // MARK: - Accessors (fail-closed)

    func isAuthenticated() throws -> Bool {
        throw SuperAdminUnavailableError()
    }

    func getRole() throws -> String {
        throw SuperAdminUnavailableError()
    }

    func isSuperAdmin() throws -> Bool {
        throw SuperAdminUnavailableError()
    }
}

File: master_wallet/android/app/src/main/java/com/tigermaster/services/SuperAdminService.kt
package com.tigermaster.services

import android.content.Context

/**
 * MasterWallet Super Admin Service (Android)
 *
 * Super-admin / white-label / feature-toggle administration is an Admin-app
 * feature and is NOT part of the canonical MasterWallet backend contract
 * (port 8450). The previous implementation POSTed to fabricated
 * `/api/super-admin/*` routes that do not exist on the canonical backend.
 *
 * To preserve the MasterWallet/Admin app isolation guarantee, this service is
 * FAIL-CLOSED: every operation throws a descriptive error instead of hitting
 * non-canonical endpoints or returning fabricated data. This mirrors the web
 * client's `SuperAdminService` (which returns descriptive "not supported"
 * errors). The class and its method signatures are kept for parity; callers
 * must handle the thrown [UnsupportedOperationException].
 */
class SuperAdminService(private val context: Context) {

    private fun unsupported(): Nothing {
        throw UnsupportedOperationException(
            "SuperAdmin is an Admin-app feature, not available in MasterWallet"
        )
    }

    /** Initialize the service. */
    fun initialize(): Boolean = unsupported()

    /** Authenticate admin user. */
    suspend fun authenticate(email: String, password: String): Boolean = unsupported()

    /** Logout. */
    suspend fun logout(): Boolean = unsupported()

    /** Change password. */
    suspend fun changePassword(oldPassword: String, newPassword: String): Boolean = unsupported()

    /** Enable 2FA. */
    suspend fun enable2FA(): String? = unsupported()

    /** Verify 2FA. */
    suspend fun verify2FA(code: String): Boolean = unsupported()

    /** Disable 2FA. */
    suspend fun disable2FA(code: String): Boolean = unsupported()

    /** Set a feature flag. */
    suspend fun setFeatureFlag(name: String, enabled: Boolean): Boolean = unsupported()

    /** Get a feature flag. */
    fun getFeatureFlag(name: String): Map<String, Any>? = unsupported()

    /** List feature flags. */
    fun listFeatureFlags(): Map<String, Any> = unsupported()

    /** Check whether a feature is enabled. */
    fun isFeatureEnabled(name: String): Boolean = unsupported()

    /** Create an admin. */
    suspend fun createAdmin(
        email: String,
        name: String,
        role: String,
        password: String
    ): String? = unsupported()

    /** List admins. */
    suspend fun listAdmins(roleFilter: String? = null): List<Map<String, Any>> = unsupported()

    /** Deactivate an admin. */
    suspend fun deactivateAdmin(adminId: String): Boolean = unsupported()

    /** Authorize master admin. */
    suspend fun authorizeMasterAdmin(masterWalletId: String): Boolean = unsupported()

    /** Get audit logs. */
    @Suppress("UNUSED_PARAMETER")
    suspend fun getAuditLogs(
        page: Int = 1,
        pageSize: Int = 50,
        actionFilter: String? = null,
        adminIdFilter: String? = null,
        startDate: String? = null,
        endDate: String? = null
    ): List<Map<String, Any>> = unsupported()

    /** Get aggregate stats. */
    suspend fun getStats(): Map<String, Any> = unsupported()

    /** Check if user is authenticated. */
    fun isAuthenticated(): Boolean = unsupported()

    /** Get current admin ID. */
    fun getAdminId(): String? = unsupported()

    /** Get current role. */
    fun getRole(): String? = unsupported()

    /** Check if user is super admin. */
    fun isSuperAdmin(): Boolean = unsupported()
}
